controllers/construct/build_statefulset_test.go

```diff
@@ -13,18 +13,34 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func init() {
 	os.Setenv(versionUpgradeHookImageEnv, "version-upgrade-hook-image")
 }
 
+func newTestReplicaSet() mdbv1.MongoDBCommunity {
+	return mdbv1.MongoDBCommunity{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "my-rs",
+			Namespace:   "my-ns",
+			Annotations: map[string]string{},
+		},
+		Spec: mdbv1.MongoDBCommunitySpec{
+			Members: 3,
+			Version: "4.2.2",
+		},
+	}
+}
+
 func TestMultipleCalls_DoNotCauseSideEffects(t *testing.T) {
-	_ = os.Setenv(mongodbRepoUrl, "repo")
-	_ = os.Setenv(mongodbImageEnv, "mongo")
+	_ = os.Setenv(MongodbRepoUrl, "repo")
+	_ = os.Setenv(MongodbImageEnv, "mongo")
+	_ = os.Setenv(AgentImageEnv, "agent-image")
 
 	mdb := newTestReplicaSet()
-	stsFunc := buildStatefulSetModificationFunction(mdb)
+	stsFunc := BuildMongoDBReplicaSetStatefulSetModificationFunction(&mdb, mdb)
```

Comment:
[q] hmm this usage is a little weird, what if the interface was also composed of the `scale.ReplicaSetScaler` interface (since no matter what it needs to be)

Reply:
I know it looks ugly! I tried that, but on enteprise AppDB does not implement `scale.ReplicaSetScaler` as the required methods are on the opsmanager type (we need to access the `status.AppDB.Replicas` field)

Reply:
(so effectively it will look like `BuildMongoDBReplicaSetStatefulSetModificationFunction(om.Spec.AppDB, om)`

Reply:
ah right I remember now, the other option I guess is to have a `GetScaler()` function on the interface, I'm okay to stick with this though, we can adjust it later if it makes sense!

Reply:
Or we could provide the AppDB with the OM status/spec fields in enterprise.

Reply:
Can we have a `GetScaler()`? AppDB type doesn't have a right reference to OpsManager type as far as I remember. 
We can discuss this later anyway!

Reply:
sure! I was more thinking now that we could add the fields to the AppDB type and let the AppDB implement the interface (the first suggestion not the GetScaler)

Reply:
Yes that will make sense, will leave for a future PR in case :)